### ISSUE TITLE
False negative of `empty_docs`  when processing proc-macros

### ISSUE THREAD

**Issue by relaxcn** [rust-lang/rust-clippy#16382]
### Summary

False negative of `empty_docs`  when processing proc-macros.

This issue should be handled with #16317 together.

### Lint Name

empty_docs

### Reproducer

I tried this code:

```rust
// src/lib.rs:
#![warn(clippy::empty_docs)]

#[cxx::bridge]
pub mod ffi {
    unsafe extern "C++" {
        type R;

        ///
        unsafe fn my_fn(&self);
    }
}

pub fn public_my_fn(r: &ffi::R) {
    unsafe { r.my_fn() }
}
```

I expected to see this happen:

warning on empty docs.

Instead, this happened:

nothing.

### Version

```text
rustc 1.94.0-nightly (fecb335cb 2026-01-07)
binary: rustc
commit-hash: fecb335cbad3d84ef3da39191ba094e6c726a5b4
commit-date: 2026-01-07
host: x86_64-unknown-linux-gnu
release: 1.94.0-nightly
LLVM version: 21.1.8
```
